### ISSUE TITLE
confusing error in callable() conditional

### ISSUE THREAD

**Issue by tlambert03** [microsoft/pyright#506]
might not be a bug, but I'm trying to keep my project strictly typed, and not sure why I'm getting this error:

**Minimal example To Reproduce**
```python
def f(a: type) -> Union[Callable[[type], Sequence], Sequence]:
    if a:
        def h(b: type):
            return [1, 2, 3]
        return h
    else:
        return [1, 2, 3]

q = f(str)
if callable(q):
    w = q(int)  # <-- pyright thinks this is not callable
```

**Expected behavior**
i expected the line `w = q(int)` to know that `q` was callable, due to the conditional.  Yet it says "`q(int)` has type `Sequence[Unknown]` and is not callable".

**Screenshots or Code**
code is above, screenshot here:
<img width="421" alt="Screen Shot 2020-02-10 at 7 30 24 PM" src="https://user-images.githubusercontent.com/1609449/74202897-d2b3c680-4c3b-11ea-9744-0c151c003f17.png">

**VS Code extension or command-line**
see it in both CLI (v1.1.21) and VS Code extension (v1.1.21)

**Comment by erictraut** [comment]
Thanks for the issue.

Pyright is able to do "type narrowing" (i.e. eliminating some of the types in a union) based on conditional tests in the code. However, it needs special logic for each flavor of conditional. It currently doesn't have any special logic for the `callable` function, so it's not performing any narrowing in the example you provided. 

I don't consider this a bug, but it is a legitimate feature request.

In the meantime, you can use a `# type: ignore` on the line to suppress the error. Or you can insert an explicit cast.

**Comment by tlambert03** [comment]
Thanks for the explanation!

**Comment by erictraut** [comment]
Thanks for reporting this. The next version of python will include type narrowing support for the built-in "callable" function.

**Comment by erictraut** [comment]
This is implemented in Pyright 1.1.22, which I just published.

**Comment by tlambert03** [comment]
Thank you!
